### PR TITLE
fixup url detect regex

### PR DESCRIPTION
**Pull request by KristofferC**
cc @innerlee

### CODE REVIEW

File: src/REPLMode.jl [hunk 1]
```diff
@@ -154,7 +154,7 @@ end
 ################
 # REPL parsing #
 ################
-const lex_re = r"^[\?\./\+\-](?!\-) | ((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)? | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
+const lex_re = r"^[\?\./\+\-](?!\-) | (((git|ssh|http(s)?)(:(//)?))|(git@[\w\.]+))([\w\.@\:/\-~]+)(\.git)(/)? | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
```
Comment (innerlee) [inline src/REPLMode.jl]: for the url `git@gitlab-fsl.jsc.nasa.gov:dmatz/MarsGRAM2010.jl.git` on the julia issue, allow `-` in `git@bla-blah.bbb:` is ok. i.e., `((git|ssh|http(s)?)|(git@[\w\-\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`

File: test/repl.jl [hunk 2]
```diff
@@ -72,6 +72,9 @@ temp_pkg_dir() do project_path; cd(project_path) do; mktempdir() do tmp_pkg_path
     tokens = Pkg.REPLMode.tokenize("add git@github.com:JuliaLang/Example.jl.git@0.5.0")
     @test tokens[1][2] == "git@github.com:JuliaLang/Example.jl.git"
     @test repr(tokens[1][3]) == "VersionRange(\"0.5.0\")"
+    tokens = Pkg.REPLMode.tokenize("add git@gitlab.fsl.jsc.näsan.guvv.drats/URGA2010.jl.git@0.5.0")
```
Comment (innerlee) [inline test/repl.jl]: Is this using some http protocol?
or there's typo for `git@gitlab.fsl.jsc.näsan.guvv:drats/URGA2010.jl.git`

Comment (KristofferC) [reply test/repl.jl]: Yes, I wrote it wrong

**Comment by KristofferC** [comment]
Updated.

**Comment by KristofferC** [comment]
(ok now I updated for real :P)

**Pull request merged**